### DISCUSSION TITLE
`TypedDict` error when using non-standard Python types

### DISCUSSION THREAD

**Discussion by multimeric** [fastapi/fastapi#11274] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from typing import TypedDict

class Bar:
    pass

class Model(TypedDict):
    bar: Bar

app = FastAPI()

@app.get("/")
def foo() -> Model:
    return {
        "bar": Bar()
    }
```

### Description

A `TypedDict` that contains non-standard Python types (in this case I use `Bar` to simulate this), results in the following error:

```
fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that <class '__main__.Model'> is a valid Pydantic field type. If you are using a return type annotation that is not a valid Pydantic field (e.g. Union[Response, dict, None]) you can disable generating the response model from the type annotation with the path operation decorator parameter response_model=None. Read more: https://fastapi.tiangolo.com/tutorial/response-model/
```

My guess is that fastapi is trying to convert `Model` into a pydantic `BaseModel` here, which fails because `arbitrary_types_allowed` is by default set to `False`. If it is indeed doing this, maybe it could be set to `True` instead?

### Operating System

macOS

### FastAPI Version

0.105.0

### Pydantic Version

2.6.3

### Python Version

3.12.2

**Comment by JavierSanchezCastro** [comment]
Did you try to do what the error told you?

```python3
from fastapi import FastAPI
from typing_extensions import TypedDict

class Bar():
    def __init__(self, a):
        self.a = a

class Model(TypedDict):
    bar: Bar

app = FastAPI()

@app.get("/", response_model=None)
def foo() -> Model:
    return Model(bar=Bar(a=42))
    #return {
    #    "bar": Bar(a=42)
    #}
```

Both 'return' works with the same expected output

**Reply by multimeric** [reply]
Of course this disables the error, but at the cost of the openapi schema being completely uninformative. The schema is just `{}`, which means that users will have little idea about the data shape, and any client-generation tool will produce the wrong types.

ie `GET /openapi.json` gives me:
```json
{
    "openapi": "3.1.0",
    "info": {
        "title": "FastAPI",
        "version": "0.1.0"
    },
    "paths": {
        "/": {
            "get": {
                "summary": "Foo",
                "operationId": "foo__get",
                "responses": {
                    "200": {
                        "description": "Successful Response",
                        "content": {
                            "application/json": {
                                "schema": {}
                            }
                        }
                    }
                }
            }
        }
    }
}
```

**Reply by JavierSanchezCastro** [reply]
I think the problem doesn't rely in TypedDict, but in the Bar class (since it was empty).

This is not working:

```python3
from fastapi import FastAPI
from typing_extensions import TypedDict


class Model():
    pass

app = FastAPI()

@app.get("/", response_model=Model)
def foo():
    return None
```

But this works well (also the openapi schema): 

```python3
from fastapi import FastAPI
from typing_extensions import TypedDict


class Model(TypedDict):
    a: int

app = FastAPI()

@app.get("/", response_model=Model)
def foo():
    return Model(a=42)
```

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
